File: 2021/1/solution.py
import os

def part1(numbers: list[int]) -> int:
  return sum([1 for a, b in zip(numbers, numbers[1:]) if b > a])

def part2(numbers: list[int]) -> int:
  return part1([sum(numbers[i-2:i+1]) for i in range(2, len(numbers))])

if __name__ == "__main__":
  with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), 'input.txt'),'r') as file:
    numbers = [int(num) for num in file]

  print(part1(numbers))
  print(part2(numbers))

File: 2021/1/solution.js
const fs = require("fs");
const path = require("path");

const input = fs
  .readFileSync(path.join(__dirname, "input.txt"), "utf-8")
  .split("\n")
  .map(Number);

const part1 = input.reduce((a, b, i) => a + (i > 0 && b > input[i - 1]), 0);

console.log(part1);

// let sumA = input[0] + input[1] + input[2];
// let sumB = input[1] + input[2] + input[3];

// let part2 = sumB > sumA ? 1 : 0;

// for (let i = 4; i < input.length; i++) {
//   sumA += input[i - 1] - input[i - 4];
//   sumB += input[i] - input[i - 3];
//   part2 += sumB > sumA;
// }

// only need to compare input[i] and input[i - 3]
let part2 = 0;
for (let i = 3; i < input.length; i++) {
  part2 += input[i] > input[i - 3];
}

console.log(part2);
